content/en/blog/_posts/2024-08-19-pod-failure-policy-for-jobs-goes-ga.md

```diff
@@ -0,0 +1,173 @@
+- `FailJob`: Fail the entire Job and terminate all running Pods.
+- `FailIndex`: Fail the index corresponding to the failed Pod
+  [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index),
+- `Count`: Count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+  This is the default behavior.
+
+When Pod failures occur in a running Job, Kubernetes matches the
+failed Pod status against the list of Pod failure policy rules, in the specified
+order, and takes the corresponding actions for the first matched rule.
+
+### Kubernetes-initiated Pod disruptions
+
+To allow matching Pod failure policy rules against failures caused by
+disruptions initiated by Kubernetes, this feature introduces the `DisruptionTarget`
+Pod condition.
+
+This condition is added by Kubernetes when a Pod fails because of a retriable
+[disruption scenario](/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions).
+The `DisruptionTarget` condition contains one of the following reasons that corresponds to
+these disruption scenarios:
+
+- `PreemptionByKubeScheduler`: [Preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption)
+   by `kube-scheduler` to accommodate a new Pod that has a higher priority.
+- `DeletionByTaintManager` - the Pod is due to be deleted by
+   `kube-controller-manager` due to a `NoExecute` [taint](/docs/concepts/scheduling-eviction/taint-and-toleration/)
+   that the Pod doesn't tolerate.
+- `EvictionByEvictionAPI` - the Pod has been marked for
+   [eviction using the Kubernetes API](/docs/concepts/scheduling-eviction/api-eviction/).
+- `DeletionByPodGC` - the Pod is bound to a node that no longer exists, and is due to
+   be deleted by [Pod garbage collection](/docs/concepts/workloads/pods/pod-lifecycle/#pod-garbage-collection).
+- `TerminationByKubelet` - the Pod was terminated by
+  [graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown),
+  [node pressure eviction](/docs/concepts/scheduling-eviction/node-pressure-eviction/)
+  or preemption for [system critical pods](/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/).
+
+### Example
+
+The Pod failure policy snippet below demonstrates an example use:
+
+```yaml
+podFailurePolicy:
+  rules:
+  - action: Ignore
+    onPodConditions:
+    - type: DisruptionTarget
+  - action: FailJob
+    onPodConditions:
+    - type: ConfigIssue
+  - action: FailJob
+    onExitCodes:
+      operator: In
+      values: [ 42 ]
+```
+
+This example, the Pod failure policy does the following:
+
+- Ignores any failed Pods that have the built-in `DisruptionTarget`
+  condition. These Pods don't count towards Job backoff limits.
+- Fails the Job if any failed Pods have the custom user-supplied
+  `ConfigIssue` condition, which was added either by a custom controller or webhook.
+- Fails the Job if any containers exited with the reserved exit code 42
```

Comment:
nit: I am not sure if `reserved` is the correct use here. It almost seems like the process should reserve the use of the exit code.

Reply:
Yes, fully agree. Dropped the word.